app/src/main/java/chat/rocket/android/chatdetails/presentation/ChatDetailsPresenter.kt

```diff
@@ -0,0 +1,71 @@
+package chat.rocket.android.chatdetails.presentation
+
+import chat.rocket.android.chatdetails.domain.ChatDetails
+import chat.rocket.android.core.lifecycle.CancelStrategy
+import chat.rocket.android.server.domain.GetCurrentServerInteractor
+import chat.rocket.android.server.infraestructure.ConnectionManagerFactory
+import chat.rocket.android.util.extension.launchUI
+import chat.rocket.common.model.roomTypeOf
+import chat.rocket.common.util.ifNull
+import chat.rocket.core.internal.rest.getInfo
+import chat.rocket.core.model.Room
+import javax.inject.Inject
+
+class ChatDetailsPresenter @Inject constructor(
+    private val view: ChatDetailsView,
+    private val navigator: ChatDetailsNavigator,
+    private val strategy: CancelStrategy,
+    serverInteractor: GetCurrentServerInteractor,
+    factory: ConnectionManagerFactory
+) {
+    private val currentServer = serverInteractor.get()!!
+    private val manager = factory.create(currentServer)
+    private val client = manager.client
+
+    fun getDetails(chatRoomId: String, chatRoomType: String) {
+        launchUI(strategy) {
+            try {
+                view.displayDetails(roomToChatDetails(client.getInfo(chatRoomId, null, roomTypeOf(chatRoomType))))
```

Comment:
Better to use the `retryIO` here `client.getInfo(chatRoomId, null, roomTypeOf(chatRoomType))`